src/dotty/tools/dotc/reporting/Reporter.scala

```diff
@@ -206,6 +230,10 @@ abstract class Reporter extends interfaces.ReporterResult {
    */
   def errorsReported = hasErrors
 
+  private[this] var reportedFeaturesUseSites = Set[Symbol]()
```

Comment:
It would be good to have a motivation why this is needed. In fact I am not yet sure we want to suppress feature warnings per use site.